source/euler_aeos/hyperbolic_system.h

```diff
@@ -927,16 +929,14 @@ namespace ryujin
 
           const auto U_i = U.template get_tensor<Number>(i);
           auto prec_i = precomputed.template get_tensor<Number, PT>(i);
-          auto &[p_i, gamma_min_i, s_i, eta_i] = prec_i;
+          auto &[p_i, gamma_i, gamma_min_i, s_i, eta_i] = prec_i;
 
           const unsigned int *js = sparsity_simd.columns(i) + stride_size;
           for (unsigned int col_idx = 1; col_idx < row_length;
                ++col_idx, js += stride_size) {
 
-            const auto U_j = U.template get_tensor<Number>(js);
             const auto prec_j = precomputed.template get_tensor<Number, PT>(js);
-            auto &[p_j, gamma_min_j, s_j, eta_j] = prec_j;
-            const auto gamma_j = surrogate_gamma(U_j, p_j);
```

Comment:
We have to balance this recomputation of `surrogate_gamma(U_j, p_j)` with the increased pressure on memory access. Unfortunately, I do see a roughly 0.5% performance regression in 2D and 3D with this change. So it is not making things faster.